error when deploying application is not shown to user

```python-traceback
Jun 16 21:43:25 brucewillis conjure-up/bundle-observable-kubernetes: [ERROR] __init__.py:19 - Showing dialog for exception:
Traceback (most recent call last):
  File "/home/ubuntu/conjure-up/conjure/juju.py", line 381, in do_add_all
    params=params)
  File "/home/ubuntu/conjure-up/macumba/v2.py", line 113, in _request
    'Params': params})
  File "/home/ubuntu/conjure-up/macumba/v2.py", line 124, in call
    return self.receive(req_id, timeout)
  File "/home/ubuntu/conjure-up/macumba/api.py", line 126, in receive
    raise ServerError(res['Error'], res)
macumba.errors.ServerError: (ServerError(...), 'application "kibana" not found')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/ubuntu/conjure-up/conjure/juju.py", line 31, in _decorator
    return f(*args, **kwargs)
  File "/home/ubuntu/conjure-up/conjure/juju.py", line 384, in do_add_all
    exc_cb(e)
  File "/home/ubuntu/conjure-up/conjure/controllers/deploy/gui.py", line 26, in __handle_exception
    app.ui.show_exception_message(exc)
  File "/home/ubuntu/conjure-up/conjure/ui/__init__.py", line 17, in show_exception_message
    self.frame.body = ErrorView(ex.args[0])
  File "/home/ubuntu/conjure-up/ubuntui/views/error.py", line 23, in __init__
    Padding.center_80(Color.error_major(Text(error))),
  File "/home/ubuntu/conjure-up/conjure-dev/lib/python3.5/site-packages/urwid/widget.py", line 836, in __init__
    self.set_text(markup)
  File "/home/ubuntu/conjure-up/conjure-dev/lib/python3.5/site-packages/urwid/widget.py", line 879, in set_text
    self._text, self._attrib = decompose_tagmarkup(markup)
  File "/home/ubuntu/conjure-up/conjure-dev/lib/python3.5/site-packages/urwid/util.py", line 375, in decompose_tagmarkup
    tl, al = _tagmarkup_recurse(tm, None)
  File "/home/ubuntu/conjure-up/conjure-dev/lib/python3.5/site-packages/urwid/util.py", line 416, in _tagmarkup_recurse
    raise TagMarkupException("Invalid markup element: %r" % tm)
urwid.util.TagMarkupException: Invalid markup element: ServerError(ServerError(...), 'application "kibana" not found')


```

--- a/bundleplacer/charmstore_api.py
+++ b/bundleplacer/charmstore_api.py
@@ -246,6 +246,7 @@ class MetadataController:
         if not self.loaded():
             self.info_callbacks.append((charm_name, cb))
             return None
+        cb(self.charm_info[charm_name])
         return self.charm_info[charm_name]
 
     def get_readme(self, short_charm_id, cb):

--- a/conjure/controllers/deploy/gui.py
+++ b/conjure/controllers/deploy/gui.py
@@ -19,11 +19,13 @@ this.bundle_filename = None
 this.bundle = None
 this.services = []
 this.svc_idx = 0
+this.showing_error = False
 
 
 def __handle_exception(tag, exc):
     utils.pollinate(app.session_id, tag)
     app.ui.show_exception_message(exc)
+    this.showing_error = True
 
 
 def finish(single_service=None):
@@ -61,7 +63,8 @@ def finish(single_service=None):
 
 
 def render():
-
+    if this.showing_error:
+        return
     if not this.bundle:
         this.bundle_filename, this.bundle, this.services = get_bundleinfo()
 

--- a/conjure/juju.py
+++ b/conjure/juju.py
@@ -345,14 +345,17 @@ def deploy_service(service, msg_cb=None, exc_cb=None):
             if exc_cb:
                 exc_cb(e)
 
+    def enqueue_deploy(new_service_info=None):
+        async.submit(partial(do_deploy, new_service_info),
+                     exc_cb,
+                     queue_name=JUJU_ASYNC_QUEUE)
+
     if service.csid.rev == "":
         id_no_rev = service.csid.as_str_without_rev()
         mc = app.metadata_controller
-        mc.get_charm_info(id_no_rev, do_deploy)
+        mc.get_charm_info(id_no_rev, enqueue_deploy)
     else:
-        async.submit(do_deploy,
-                     exc_cb,
-                     queue_name=JUJU_ASYNC_QUEUE)
+        enqueue_deploy()
 
 
 def set_relations(services, msg_cb=None, exc_cb=None):

